staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4174,6 +4194,12 @@ type PodStatus struct {
 	// +featureGate=InPlacePodVerticalScaling
 	// +optional
 	Resize PodResizeStatus `json:"resize,omitempty" protobuf:"bytes,14,opt,name=resize,casttype=PodResizeStatus"`
+
+	// Status of resource claims.
+	// +listType=atomic
+	// +featureGate=DynamicResourceAllocation
+	// +optional
+	ResourceClaimStatuses []PodResourceClaimStatus `json:"resourceClaimStatuses,omitempty" protobuf:"bytes,15,rep,name=resourceClaimStatuses"`
```

Comment:
This cannot be `+listType:map` because the field that could serve as key, the `sourceRef`, is a struct and those cannot be keys.

It also cannot be a `listType=set` because the strategic merging cannot be used to patch just this one field. The apiserver then preserves old and new entries (I tried...).

This is okay. There's a single writer (the controller) and it can ensure that the field gets updated atomically.